.github/workflows/deploy_lambda_dev.yml

```
@@ -0,0 +1,23 @@
+name: Deploy Lambda to Dev
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: Branch to deploy
+        required: true
+        default: master
+
+jobs:
+  setup-job:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.ref }}
+
+  build_deploy_lambda:
```

```suggestion
  build_deploy_lambda:
    needs: [setup-job]
```
I believe this is needed. If not, touch base just so I know why not.

---

Yep, good catch. Keeps that checkout the first step that is run and then this one. We need that checkout in order to pull the actions in correctly and have them run.